actionview/lib/action_view/structured_event_subscriber.rb

```diff
@@ -0,0 +1,92 @@
+# frozen_string_literal: true
+
+require "active_support/structured_event_subscriber"
+
+module ActionView
+  class StructuredEventSubscriber < ActiveSupport::StructuredEventSubscriber # :nodoc:
+    VIEWS_PATTERN = /^app\/views\//
+
+    def initialize
+      @root = nil
+      super
+    end
+
+    def render_template(event)
+      emit_event("action_view.render_template",
+        identifier: from_rails_root(event.payload[:identifier]),
+        layout: from_rails_root(event.payload[:layout]),
+        duration: event.duration.round(1),
+        gc: event.gc_time.round(1),
+      )
+    end
+
+    def render_partial(event)
+      emit_event("action_view.render_partial",
+        identifier: from_rails_root(event.payload[:identifier]),
+        layout: from_rails_root(event.payload[:layout]),
+        duration: event.duration.round(1),
+        gc: event.gc_time.round(1),
+        cache_hit: event.payload[:cache_hit],
+      )
+    end
+
+    def render_layout(event)
+      emit_event("action_view.render_layout",
+        identifier: from_rails_root(event.payload[:identifier]),
+        duration: event.duration.round(1),
+        gc: event.gc_time.round(1),
+      )
+    end
+
+    def render_collection(event)
+      emit_event("action_view.render_collection",
+        identifier: from_rails_root(event.payload[:identifier] || "templates"),
+        layout: from_rails_root(event.payload[:layout]),
+        duration: event.duration.round(1),
+        gc: event.gc_time.round(1),
+        cache_hits: event.payload[:cache_hits],
+        count: event.payload[:count],
+      )
+    end
+
+    module Utils # :nodoc:
+      private
+        def from_rails_root(string)
+          return unless string
+
+          string = string.sub(rails_root, "")
+          string.sub!(VIEWS_PATTERN, "")
+          string
+        end
+
+        def rails_root # :doc:
+          @root ||= "#{Rails.try(:root)}/"
```

Comment:
This might be a bit undesirable. The log subscriber waits until a logger is set, but the structured event subscriber doesn't need to so things get triggered early enough that Rails sometimes doesn't have root defined yet. Generally, we should not be referencing Rails at all here, and I want to work on injecting the root here in a cleaner way when I consolidate the notification subscribers. But, I can try to to do that earlier if we feel strongly about this being bad.